src/core/server/integration_tests/saved_objects/service/lib/repository_with_proxy.test.ts

```diff
@@ -93,29 +93,40 @@ describe('404s from proxies', () => {
       ? parseInt(process.env.TEST_PROXY_SERVER_PORT, 10)
       : 5698;
 
+    // Setup kibana configured to use proxy as ES backend
+    root = createRootWithCorePlugins({
+      elasticsearch: {
+        hosts: [`http://${esHostname}:${proxyPort}`],
+      },
+      migrations: {
+        skip: false,
+      },
+    });
+    await root.preboot();
+    const setup = await root.setup();
+    registerSOTypes(setup);
+
     // Setup custom hapi hapiServer with h2o2 plugin for proxying
     hapiServer = Hapi.server({
       port: proxyPort,
     });
 
+    const mainIndex = getVersionedKibanaIndex();
     await hapiServer.register(h2o2);
     // register specific routes to modify the response and a catch-all to relay the request/response as-is
+    declareGetRoute(hapiServer, esHostname, esPort, mainIndex);
+    declareDeleteRoute(hapiServer, esHostname, esPort, mainIndex);
+    declarePostUpdateRoute(hapiServer, esHostname, esPort, mainIndex);
 
-    allCombinationsPermutations(
-      ALL_SAVED_OBJECT_INDICES.map((indexPattern) => `${indexPattern}_${pkg.version}`)
-    )
-      .map((indices) => indices.join(','))
-      .forEach((kbnIndexPath) => {
-        declareGetRoute(hapiServer, esHostname, esPort, kbnIndexPath);
-        declareDeleteRoute(hapiServer, esHostname, esPort, kbnIndexPath);
-        declarePostUpdateRoute(hapiServer, esHostname, esPort, kbnIndexPath);
-
-        declareGetSearchRoute(hapiServer, esHostname, esPort, kbnIndexPath);
-        declarePostSearchRoute(hapiServer, esHostname, esPort, kbnIndexPath);
-        declarePostPitRoute(hapiServer, esHostname, esPort, kbnIndexPath);
-        declarePostUpdateByQueryRoute(hapiServer, esHostname, esPort, kbnIndexPath);
-        declareIndexRoute(hapiServer, esHostname, esPort, kbnIndexPath);
-      });
+    declareGetSearchRoute(hapiServer, esHostname, esPort, mainIndex);
+    declarePostSearchRoute(hapiServer, esHostname, esPort, mainIndex);
+    declarePostPitRoute(hapiServer, esHostname, esPort, mainIndex);
+    declareIndexRoute(hapiServer, esHostname, esPort, mainIndex);
+
+    // the deleteByNamespace performs an updateByQuery under the hood.
+    // It targets all SO indices that have namespace-aware types
+    const nsIndices = getIndicesWithNamespaceAwareTypes(setup.savedObjects.getTypeRegistry());
```

Comment:
Previously we were running a very costly operation, registering thousands of routes, when 99% of requests target `.kibana` solely.
The only case where multiple indices were specified in the HTTP request path is with namespace-aware indices.
The test was timing out due to the new SO index, which added a lot more permutations.
Hence, I used the opportunity to simplify it.
The test should now be noticeably ligher/faster.